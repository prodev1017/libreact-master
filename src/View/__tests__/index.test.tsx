import {createElement as h} from 'react';
import {render} from 'react-dom';
import renderer from 'react-test-renderer';
import {View} from '..';

describe('<View/>', () => {
  it('renders without crashing', () => {
    render(<View />, document.createElement('div'));
  });

  it('renders nothing by default', () => {
    expect(renderer.create(<View />).toJSON()).toMatchSnapshot();
  });

  it('renders its children', () => {
    const jsx = <View><div>children....</div></View>;

    expect(renderer.create(jsx).toJSON()).toMatchSnapshot();
  });

  it('renders child function as children', () => {
    const jsx = <View>{() => <div>children....</div>}</View>;

    expect(renderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
