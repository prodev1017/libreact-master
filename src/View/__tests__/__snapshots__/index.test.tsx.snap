// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View/> renders child function as children 1`] = `
<div>
  children....
</div>
`;

exports[`<View/> renders its children 1`] = `
<div>
  children....
</div>
`;

exports[`<View/> renders nothing by default 1`] = `null`;
